Sync CEPH working branch part 3

xarses opened the pull request:
This will sync the current CEPH working branch

Works with Mirantis/fuelweb#1611. Now uses checkboxes to enable cinder/glance/object parts.

What works:
- MON
- OSD
- Cinder
- Glance
- Nova

Still WIP:
- rados-gw

Commit:
re-enable ceph; create condition for openstack::img::cirros

Commit:
Fix ceph glance/cinder bugs

remove scp commands, should be handled by Exec['ceph-deploy init config']
 and ceph auth get-or-create commands
remove dep-cycle in cluster_[ha/simple]
add unless to ceph-deploy mon create
overide key_val_seperator for ceph_conf
move secret.xml to /root

Commit:
update seperator regex in inifille

Commit:
change ssh_config file content back to doublequote

Commit:
move ceph_conf inside pirmary_mon block to prevent duplicate lines

Commit:
add additional test to ceph::[cinder|glance]

Commit:
[PRD-1552] Enable and configure rados-gw

distroize manafests
remove dupe declrations
enable ceph::keystone and ceph::radosgw for monitor roles
add service for radosgw
add service for ceph
move libnss to init.pp
move file /etc/ceph/nss to init.pp
fix keystone_endpoint region
move rados ceph_conf parts to radosgw
add paramater for enable and use_ssl for radosgw and keystone

Commit:
move cinder_config and glance_api_config changes from ceph to cinder and glance modules

Commit:
refactor Ceph init.pp

 * extract ceph.conf handling into ceph::conf
 * rename ceph::deploy to ceph::mds (and leave it disabled)
 * move libnss handling to ceph::libnss class in params.pp
 * move ceph service definition to ceph::params
 * simplify role-dependent declarations and ordering rules
 * include ceph::cinder for cinder nodes, but don't include ceph classes
   by default for unrecognized node roles
 * install python-pushy instead of pushy in Debian-based systems
 * minor formatting fixes to calm down puppet-lint

Commit:
force radiosgw in cluster_ha and cluster_simple

Commit:
[PRD-1552] fixes

distroize rgw.conf.erb
fix ceph::ssh package if defined
fix ordering of keystone
fix ordering of ceph service

Commit:
refactor Ceph mon.pp

 * drop keyring file declarations, check for ceph-create-keys instead of
   running ceph-deploy gatherkeys to check for Ceph cluster quorum
 * move Ceph client ACLs definition to cinder.pp and glance.pp
 * extract c_pools subclass into a proper ceph::osd_pools class
 * parametrize Cinder and Glance pool names explicitly and separately
   instead of accessing them by indices in a list
 * drop the checks for $::glance_api_conf and $::cinder_conf

Commit:
refactor Ceph osd.pp

 * replace fine-grained require 'ceph-deploy init config' with a
   coarse-graned chaining from ceph::conf to ceph::osd
 * extract $osd_devices check and cross-manifest relations to init.pp
 * add missing include ceph::mon to ceph::controller

Commit:
Use plain partitions for Ceph journals

Commit:
refactor Ceph nova_compute.pp

 * declare ceph::nova_compute only for compute role, it's useless in
   ceph::controller
 * drop TODO for moving the RBD secret.xml to mktemp (it's much safer to
   keep it in /root/, and virsh is unable to accept it on stdin so we
   can't get rid of it altogether)
 * move external relationship declarations to init.pp
 * drop nova-compute service redefinition, include nova::compute from
   init.pp instead

Commit:
Allow swift to be disabled

Group swift var decl together
Group swift classes in contoller role
Add if use_swift vars to both

Commit:
bugfixes for Ceph refactoring

 * merge conf_new and conf_pull back into ceph::conf
 * merge osd_pools back into ceph::mon
 * ceph-deploy gatherkeys has to run on primary mon, too
 * declare ceph service in init.pp instead of params.pp
 * don't re-run ceph-deploy osd prepare if the OSD devices are already
   listed in /proc/mounts
 * move ceph::libnss to its own file, include it in init.pp

Commit:
[PRD-1543] Add ceph module to fuel site

Add storage_hash
create tests to see if ceph or swift parts should be enabled from the
 states in storage_hash
set glance_backend from storage_hash
correct some use_ceph tests

Commit:
adapt osnailyfacter to storage backend checkboxes

Commit:
change ceph chain in cluster_ha/simple

Commit:
[PRD-1543] enable ceph in site part 2.

Add move ceph::cinder to cinder::volumes:ceph
* add test for manage_volumes to except iscsi or ceph
* add osnaily test to set manage_volumes
Move ceph::glance to glance::backend::ceph

Commit:
Remove dupe lines for syslog (in vhost template already)

Commit:
[PRD-1552] Fix rados-gw issues

move radosgw to another port to prevent conflict with horizon
fix radosgw module to use parent ceph params
change from using mgmt ip to public in cluster_
drop un-needed dep on nova::compute
chain radosgw after ceph::mon

Commit:
Fix cinder chains

add chain between db sync and:
* cinder-volume
* cinder-scheduler
* cinder-api

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/215/

mihgen commented on deployment/puppet/ceph/manifests/apt.pp:
```diff
@@ -19,9 +20,9 @@
   }
   apt::source { 'radosgw-apache2':
-    location => "http://gitbuilder.ceph.com/apache2-deb-precise-x86_64-basic/ref/master/",
+    location => 'http://gitbuilder.ceph.com/apache2-deb-precise-x86_64-basic/ref/master/',
```
Why do we need this? Is deployment still going to work if master node / slaves will not have access to Internet?

xdel replied:
Just ask CI team, but current Ubuntu does have all necessary Ceph-related packages, so you may drop it.

xarses replied:
This is only a syntax checker fix for a file we don't actively use. It's only kept for internal use.

xdel commented on deployment/puppet/ceph/manifests/nova_compute.pp:
```diff
@@ -1,37 +1,23 @@
+    content => template('ceph/secret.erb')
   }
+
+  exec {'Set Ceph RBD secret for Nova':
```
BTW qemu WILL expose key in the cmdline so if we care about so-called OpenStack security we must use key with lower than admin privileges for VM operations. In real world nobody will explore such vuln in the wild because shell access to the any node in the env gives a plenty of other possibilities to raise itself in mean of privileges.

xdel commented on deployment/puppet/ceph/templates/rgw.conf.redhat.erb:
```diff
@@ -0,0 +1,21 @@
+<VirtualHost *:<%= @rgw_port %>>
+  ServerName <%= @fqdn %>
+  DocumentRoot /var/www
```
Nope, please add another hier level down. /var/www/ should not serve only one root alone by its purpose.

xdel commented on deployment/puppet/ceph/templates/rgw.conf.ubuntu.erb:
```diff
@@ -1,6 +1,6 @@
-<VirtualHost *:80>
+<VirtualHost *:<%= @rgw_port %>>
   ServerName <%= @fqdn %>
   DocumentRoot /var/www
```
and everywhere too, please add /s3/

xdel commented on deployment/puppet/osnailyfacter/manifests/cluster_ha.pp:
```diff
@@ -339,12 +369,22 @@
+      class { 'swift::keystone::auth':
+        password         => $swift_hash[user_password],
+        public_address   => $public_vip,
+        internal_address => $management_vip,
```
Please make a note in haproxy manifest that under no circumstances apache balancing should be changed to hash or rr mode from active+backup.

xdel commented:
By the way, please add remark considering ultimate NTP stability of the controllers. It needs throughout checking during QA stage.

Commit:
Fix param and indent alignment in modified blocks

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/227/

xarses commented:
retest this please

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/231/

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/232/

Commit:
Fix more radosgw issues

Add missing Listen directive
Add missing chain from Ceph_conf to radosgw service
Change default rgw data directory
distroize rgw-user

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/234/

xdel commented:
everything is fine, ++
let`s wait for f-web part approval

Commit:
Change ceph_conf params to use _ seperators instead of space

Commit:
Clean up from rebase on master

Fix conflicts from rebase in yaml facts

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/250/

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/261/

Commit:
Run all late_command parted commands in-target

Commit:
Label filesystems in Ubuntu

Commit:
Refactor radosgw parts

refactor radosgw to align all chains
unify rgw.conf using mod_fastcgi

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/264/

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/265/

Commit:
cleanup radosgw dependencies

 * move Ceph_file spaceship up the chain to avoid circular dependency
 * move the rest of package->file requires into the main chain
 * drop if(enabled): don't include the class if you don't need it
 * typos: s3gw.fcgi.erb 766->755, keyring 650->640, radiosgw->radosgw
 * drop the completed ssh, cinder, and glance TODO comments
 * syntax checker warnings fixes

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/266/

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/290/

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/292/

Commit:
Change pmanager msdos -> gpt; Fix dangling } in cluster_simple

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/294/

aglarendil commented:
+2
